src/isar/state_machine/state_machine.py

```diff
@@ -92,30 +180,106 @@ def __init__(
         self.transitions_log_length = transitions_log_length
         self.transitions_list: Deque[States] = deque([], self.transitions_log_length)
 
+    #################################################################################
+    # Transition Callbacks
+    def _step_initiated(self) -> None:
+        self.current_step.status = StepStatus.InProgress
+        self.publish_step_status()
+        self.logger.info(
+            f"Successfully initiated "
+            f"{type(self.current_step).__name__} "
+            f"step: {str(self.current_step.id)[:8]}"
+        )
+
+    def _pause(self) -> None:
+        self.paused = True
+
+    def _resume(self) -> None:
+        self.logger.info("Resuming mission!")
+        self.paused = False
+        self.current_mission.status = MissionStatus.InProgress
+        self.current_task.status = TaskStatus.InProgress
+        self.publish_mission_status()
+        self.publish_task_status()
+        self.queues.resume_mission.output.put(True)
+        self.current_task.reset_task()
+        self.update_current_task()
```

Comment:
Is this call necessary? It should just skip the entire function anyway

Reply:
Agree